teams/vsp/teams/tools/frontend/cypress-migration-guide.md

```diff
@@ -0,0 +1,242 @@
+
+## Writing Tests <a name="writing-tests"></a>
+### Test Structure <a name="test-structure"></a>
+The test structure of Cypress should feel familiar. Cypress uses mocha's bdd syntax which is what we use for our unit tests. Note that because `it()` blocks are their own tests, each `it()` block should be independent of others in the same testing suite. Everything executed in the browser needs to be inside of an `it()` block. Visit [here](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Test-Structure) for more context.
+
+### Visiting a Page <a name="visiting-a-page"></a>
+When [visiting a page](https://docs.cypress.io/api/commands/visit.html#Syntax), you do not need to be concerned about the `baseUrl`. Cypress's configuration file takes care of this. So rather than grabbing the `baseUrl` from the helpers in nightwatch:
+
+```javascript
+client.openUrl(`${E2eHelpers.baseUrl}/health-care/apply/application`)
+```
+
+You can instead enter the page directly:
+
+```javascript
+cy.visit('health-care/apply/application')
+```
+
+### Interacting with Page Elements <a name="interacting-with-page-elements"></a>
+Cypress has a large api that allows for easy interacting with elements. The most common tasks you will probably need to convert from nightwatch tests are clicking, selecting elements from dropdowns, and entering input. Shown below is an overview of how to perform these tasks in cypress.
+
+#### Clicking
+Nightwatch:
+```javascript
+client.click('.form-panel .usa-button-primary');
+```
+Cypress:
+```javascript
+cy.get('.form-panel .usa-button-primary').click();
+```
+
+#### Selecting From Dropdown
+Nightwatch:
+```javascript
+client.selectDropdown('root_veteranAddress_country', data.veteranAddress.country)
+```
+Cypress:
+```javascript
+cy.get('#root_veteranAddress_state').select(testData.veteranAddress.state)
+```
+
+#### Entering Data
+Nightwatch:
+```javascript
+client.fill('input[name="root_firstName"]', data.veteranFullName.first)
+```
+Cypress:
+```javascript
+cy.get('#root_firstName').type(testData.veteranFullName.first)
+```
+
+Please visit [here](https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Actionability) for more information about interacting with elements. For additional ways to interact with the DOM we have also included the [cypress testing library](https://testing-library.com/docs/cypress-testing-library/intro) in the repo.
+
+### Custom Commands <a name="custom-commands"></a>
+
+Some custom nightwatch commands located in `src/platform/testing/e2e/nightwatch-commands` were converted to cypress commands, and can be found in `src/platform/testing/e2e-cypress/support`. Many of them did not need to be converted because cypress supports their functionality natively. These custom commands can be used directly on `cy`. For example:
+
+```javascript
+cy.fillDate('root_dob', testData.veteranDateOfBirth);
```

Comment:
Just realized this is also used as an example for the Accessibility section later... So not sure what a good example would be, since we're not using `fillDate`. Maybe it's fine to repeat.

Reply:
Good catch. I think it's fine to repeat.